1/mdav2/frequencies.py

```diff
@@ -0,0 +1,91 @@
+import sys
+import numpy as np
+sys.path.append('../../extra-files')
+sys.path.append('../../0/mdav2')
+import molecule
+import math
+
+ccms = 3E10 #speed of light in centimeters per second
+eh = 4.36E-18 #J/hartree
+ao = 5.29E-11 #m/br
+u = 1.66E-27 #kg/amu
+
+def readhessian(filename):
+    #opens and reads a hessian matrix from the given filename
+    with open(filename, 'r') as hessianfile:
+        hessianlines = hessianfile.readlines()
+
+
+    hessianmatrix = np.zeros((len(hessianlines),len(hessianlines)))
+    for line in range(0,len(hessianlines)):
+        hessianmatrix[line] = hessianlines[line].split()
+    return hessianmatrix
+
+def frequencies(molecule,hessian):
+    #calculates normal modes and frequencies from a Hessian matrix. Takes a molecule file and a \
+    # 3*natom X 3*natom hessian numpy array as input.
+    weightedhessian = np.zeros(hessian.shape)
+    uwvectors = np.zeros(hessian.shape)
+    frequencies = []
+    sortedfreqs = []
+    normalmodes = []
+
+    #TODO:find a better way to make this matrix
+    for A in range(0,molecule.natom * 3):
+        for B in range(0, molecule.natom * 3):
+            weightedhessian[A][B] = hessian[A][B]/math.sqrt(molecule.masses[math.floor(A/3.)]*molecule.masses[math.floor(B/3.)])
+
+    evals,evecs = np.linalg.eig(weightedhessian)
```

Comment:
It's recommended to use eigh when you can. It saves time (and in some cases, reduces errors!) when diagonalizing a Hermitian matrix.